src/main.py

```diff
@@ -146,10 +156,20 @@ def run_ci(args):
     return ci_pr(pull)
 
 
-def _get_changelog_entries(repo, start_sha):
+def _get_changelog_entries(repo, start_sha, end_sha):
     entries = []
 
+    passed_end_sha = False
+    if not end_sha:
+        passed_end_sha = True
```

Comment:
Wouldn't this mean that `end_sha` is blank and therefore `passed_end_sha` would be false?

Reply:
If `end_sha` is blank, then there's no need to skip commits waiting for the end_sha, so `passed_end_sha` is set to True so that the continue later on is never hit. I'm not super thrilled about this implementation, so open to alternatives or different naming schemes. I suppose I could remove this and below change the line to 

```
        if not passed_end_sha and end_sha is not None:
            continue
```

Not sure if that's easier or harder to read, though

Reply:
Ah, I mistook `passed` to mean that an `end_sha` was passed in, but I'm following the logic now. This is probably a bit pedantic, but maybe `after_end_sha` or `past_end_sha` would be more clear? I'm not against leaving it either.

Reply:
@dhaworth I reworked the implementation to try to make it easier to follow. Can you take another look and let me know if it makes sense?